.github/workflows/release.yml

```diff
@@ -0,0 +1,53 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+
+jobs:
+  release-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Update Release PR
+        uses: google-github-actions/release-please-action@v3
+        id: release
+        with:
+          release-type: ruby
+          package-name: next_rails
+          version-file: "lib/next_rails/version.rb"
+          changelog-types: '[{"type":"feat","section":"Features","hidden":false}, {"type":"fix","section":"Bug Fixes","hidden":false}, {"type":"chore", "section":"Miscellaneous","hidden":false}]'
+      - name: Checkout
+        uses: actions/checkout@v3
+        if: ${{ steps.release.outputs.release_created }}
+      - name: Set Up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundle-cache: true
+        if: ${{ steps.release.outputs.release_created }}
+      - name: Bundle Install
+        run: bundle install
+        if: ${{ steps.release.outputs.release_created }}
+      - name: Publish to GitHub
+        run: |
+          mkdir -p $HOME/.gem
+          touch $HOME/.gem/credentials
+          chmod 0600 $HOME/.gem/credentials
+          printf -- "---\n:github: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+          gem build *.gemspec
+          gem push --KEY github --host https://rubygems.pkg.github.com/${OWNER} *.gem
+        env:
+          GEM_HOST_API_KEY: "Bearer ${{secrets.GITHUB_TOKEN}}"
+          OWNER: ${{ github.repository_owner }}
+        if: ${{ steps.release.outputs.release_created }}
+      - name: Publish to RubyGems
+        run: |
+          mkdir -p $HOME/.gem
+          touch $HOME/.gem/credentials
+          chmod 0600 $HOME/.gem/credentials
+          printf -- "---\n:rubygems_api_key: ${GEM_HOST_API_KEY}\n" > $HOME/.gem/credentials
+          gem build *.gemspec
+          gem push *.gem
+        env:
+          GEM_HOST_API_KEY: "${{ secrets.RUBYGEMS_AUTH_TOKEN }}"
```

Comment:
The last step is we need to generate a token in the RubyGems account (you can scope the token per gem or global) and add it to the repo under `next_rails > settings > secrets > actions > repository secrets` with the following name `RUBYGEMS_AUTH_TOKEN`

![image](https://user-images.githubusercontent.com/7331511/177388324-a86826fa-3d7e-41e0-b7f9-3c3e23594dac.png)